src/TestFramework/Factory.php

```diff
@@ -67,60 +48,73 @@ public function __construct(
-        XmlConfigurationHelper $xmlConfigurationHelper,
-        string $jUnitFilePath,
         InfectionConfig $infectionConfig,
         VersionParser $versionParser
     ) {
         $this->tmpDir = $tmpDir;
         $this->configLocator = $configLocator;
-        $this->xmlConfigurationHelper = $xmlConfigurationHelper;
         $this->projectDir = $projectDir;
-        $this->jUnitFilePath = $jUnitFilePath;
         $this->infectionConfig = $infectionConfig;
         $this->versionParser = $versionParser;
     }
 
+    private function checkClassExists(string $className, string $classType): self
+    {
+        if (!class_exists($className)) {
+            throw new \LogicException('Framework ' . $classType . ' Should Be Named ' . $className);
+        }
+
+        return $this;
+    }
+
     public function create(string $adapterName, bool $skipCoverage): AbstractTestFrameworkAdapter
     {
-        if ($adapterName === TestFrameworkTypes::PHPUNIT) {
-            $phpUnitConfigPath = $this->configLocator->locate(TestFrameworkTypes::PHPUNIT);
-            $phpUnitConfigContent = file_get_contents($phpUnitConfigPath);
-            \assert(\is_string($phpUnitConfigContent));
-
-            return new PhpUnitAdapter(
-                new TestFrameworkFinder(TestFrameworkTypes::PHPUNIT, $this->infectionConfig->getPhpUnitCustomPath()),
-                new InitialConfigBuilder(
-                    $this->tmpDir,
-                    $phpUnitConfigContent,
-                    $this->xmlConfigurationHelper,
-                    $this->jUnitFilePath,
-                    $this->infectionConfig->getSourceDirs(),
-                    $skipCoverage
-                ),
-                new MutationConfigBuilder($this->tmpDir, $phpUnitConfigContent, $this->xmlConfigurationHelper, $this->projectDir),
-                new ArgumentsAndOptionsBuilder(),
-                $this->versionParser
+        if (!in_array($adapterName, TestFrameworkTypes::TYPES)) {
+            throw new \InvalidArgumentException(
+                sprintf(
+                    'Invalid name of test framework. Available names are: %s',
+                    implode(', ', TestFrameworkTypes::TYPES)
+                )
             );
         }
 
-        if ($adapterName === TestFrameworkTypes::PHPSPEC) {
-            $phpSpecConfigPath = $this->configLocator->locate(TestFrameworkTypes::PHPSPEC);
+        $configPath = $this->configLocator->locate($adapterName);
 
-            return new PhpSpecAdapter(
-                new TestFrameworkFinder(TestFrameworkTypes::PHPSPEC),
-                new PhpSpecInitialConfigBuilder($this->tmpDir, $phpSpecConfigPath, $skipCoverage),
-                new PhpSpecMutationConfigBuilder($this->tmpDir, $phpSpecConfigPath, $this->projectDir),
-                new PhpSpecArgumentsAndOptionsBuilder(),
-                $this->versionParser
-            );
+        $baseNamespace = __NAMESPACE__ . '\\' . $adapterName . '\\';
```

Comment:
I prefer the old if, else statements, it does add mroe boilerplate code, but in all reality there won't be much test frameworks being added. This is too 'magical', and makes is harder to reason about what is going on.